Packages/MIES/MIES_SweepFormula.ipf

```diff
@@ -4045,6 +4049,37 @@ static Function/WAVE SF_OperationRange(variable jsonId, string jsonPath, string
 	return SFH_GetOutputForExecutorSingle(range, graph, SF_OP_RANGE, dataType = SF_DATATYPE_RANGE)
 End
 
+/// concat(array0, array1, array2, ...)
+static Function/WAVE SF_OperationConcat(variable jsonId, string jsonPath, string graph)
+
+	variable numArgs, i, err, majorType
+	variable constantDataType
+	string refDataType, dataType, wvNote
+
+	numArgs = SFH_CheckArgumentCount(jsonId, jsonPath, SF_OP_CONCAT, 1)
+
+	WAVE result = SFH_GetArgumentAsWave(jsonId, jsonpath, graph, SF_OP_CONCAT, 0, copy = 1, singleResult = 1, wvNote = wvNote)
+	majorType = WaveType(result, 1)
+
+	refDataType      = JWN_GetStringFromNote(wvNote, SF_META_DATATYPE)
+	dataType         = refDataType
+	constantDataType = !IsEmpty(refDataType)
+	Note/K result
+
+	AssertOnAndClearRTError()
+	for(i = 1; i < numArgs; i += 1)
+		WAVE slice = SFH_GetArgumentAsWave(jsonId, jsonpath, graph, SF_OP_CONCAT, i, singleResult = 1, wvNote = wvNote, expectedMajorType = majorType)
+		dataType         = JWN_GetStringFromNote(wvNote, SF_META_DATATYPE)
+		constantDataType = constantDataType && !CmpStr(refDataType, dataType)
+
+		Concatenate/FREE/NP {slice}, result; err = GetRTError(1)
+		SFH_ASSERT(!err, "Error concatenating waves: " + GetErrMessage(err))
```

Comment:
Just as comment:
I noticed here that you do not use substitutionOption 3 here.

I was pondering a bit if GetRTErrMessage would be the better option here:
```igorpro
Concatenate/FREE/NP {slice}, result; errMsg = GetRTErrMessage(); err = GetRTError(1)
SFH_ASSERT(!err, "Error concatenating waves: " + errMsg)
```

Reply:
Good point. Fixed.